File: app/src/main/java/igrek/todotree/domain/treeitem/TextTreeItem.java
package igrek.todotree.domain.treeitem;


public class TextTreeItem extends AbstractTreeItem {
	
	private String name;
	
	public TextTreeItem(AbstractTreeItem parent, String name) {
		super(parent);
		this.name = name;
	}
	
	public TextTreeItem(String name) {
		this(null, name);
	}
	
	@Override
	public TextTreeItem clone() {
		return new TextTreeItem(null, name).copyChildren(this);
	}
	
	@Override
	public String getDisplayName() {
		return name;
	}
	
	@Override
	public String getTypeName() {
		return "text";
	}
	
	public void setName(String name) {
		this.name = name;
	}
}

File: app/src/main/java/igrek/todotree/domain/treeitem/TextTreeItem.kt
package igrek.todotree.domain.treeitem

class TextTreeItem(parent: AbstractTreeItem?, override var displayName: String) : AbstractTreeItem(parent) {

    constructor(name: String) : this(null, name)

    override fun clone(): TextTreeItem {
        return TextTreeItem(null, displayName).copyChildren(this)
    }

    override val typeName: String
        get() = "text"

    fun setName(name: String) {
        displayName = name
    }

}